### PR TITLE
Simpler setCoords

### PR DESCRIPTION
**Pull request by asturur**
Switched from 4 transformPoint call to 2
Removed isFinite statement
changed hypotenuse calculation with something that if not faster is at least more readable

```
Math.sqrt(
            Math.pow(currentWidth / 2, 2) +
            Math.pow(currentHeight / 2, 2)),
```
with
```
(currentWidth / Math.cos(_angle)) / 2
```
removed 2 extra multiplication

### CODE REVIEW

**Comment by asturur** [comment]
I tried to measure performances of this changes, jsperf is down,
i tried on kitchensink i get different results, it looks like is twice as fast.

**Comment by asturur** [comment]
Also changed:
sqrt(2 * x^2) / 2  ===>   x * sqrt(2) / 2 ===> x * 0.707106

File: src/mixins/object_interactivity.mixin.js [hunk 1]
```diff
@@ -77,7 +77,8 @@
     _setCornerCoords: function() {
       var coords = this.oCoords,
           newTheta = degreesToRadians(45 - this.angle),
-          cornerHypotenuse = Math.sqrt(2 * Math.pow(this.cornerSize, 2)) / 2,
+          /* Math.sqrt(2 * Math.pow(this.cornerSize, 2)) / 2, */
+          cornerHypotenuse = this.cornerSize * 0.707106,
```
Comment (kangax) [inline src/mixins/object_interactivity.mixin.js]: What does `0.707106` stand for?

File: src/mixins/object_geometry.mixin.js [hunk 2]
```diff
@@ -305,10 +305,8 @@
     setCoords: function() {
       var theta = degreesToRadians(this.angle),
           vpt = this.getViewportTransform(),
-          f = function (p) {
-            return fabric.util.transformPoint(p, vpt);
-          },
-          p = this._calculateCurrentDimensions(false),
+          f = fabric.util.transformPoint,
```
Comment (kangax) [inline src/mixins/object_geometry.mixin.js]: I'm not a fan of 1-2 char names, even though I see we already had it! Can we rename it to at least `transform`?

File: src/mixins/object_geometry.mixin.js [hunk 3]
```diff
@@ -305,10 +305,8 @@
...
+          p = this._calculateCurrentDimensions(true),
```
Comment (kangax) [inline src/mixins/object_geometry.mixin.js]: Same here: `p` &rarr; `dimensions` (or at least `dim`)

File: src/mixins/object_interactivity.mixin.js (hunk 1 above)
Comment (kangax) [inline src/mixins/object_interactivity.mixin.js]: Oh, I see your explanation above. We'll definitely need a comment for that!

**Comment by asturur** [comment]
@kangax changes applied

File: src/mixins/object_interactivity.mixin.js [hunk 4]
```diff
@@ -77,7 +77,9 @@
     _setCornerCoords: function() {
       var coords = this.oCoords,
           newTheta = degreesToRadians(45 - this.angle),
-          cornerHypotenuse = Math.sqrt(2 * Math.pow(this.cornerSize, 2)) / 2,
+          /* Math.sqrt(2 * Math.pow(this.cornerSize, 2)) / 2, */
+          /* 0.707106 stands for sqrt(2)/2 */
```
Comment (kangax) [inline src/mixins/object_interactivity.mixin.js]: Wait, wouldn't this value (`0.707...`) change depending on `cornerSize`?

**Comment by asturur** [comment]
is math:

Math.sqrt(2 * Math.pow(this.cornerSize, 2)) / 2

equal to

Math.sqrt(2) * Math.sqrt(Math.pow(this.cornerSize, 2)) / 2

equal to

Math.sqrt(2) / 2 * Math.sqrt(Math.pow(this.cornerSize, 2))

equal to

Math.sqrt(2) / 2 * this.cornerSize

equal to

fixedNumber * this.cornerSize.

**Comment by kangax** [comment]
lol, of course. I missed `this.cornerSize * ...` part :)

**Pull request merged**